packages/core/notifications/core-notifications-browser-internal/src/toasts/telemetry/event_types.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { RootSchema } from '@kbn/analytics-client';
+import { EventTypeOpts } from '@kbn/core/public';
+
+export enum EventMetric {
+  TOAST_DISMISSED = 'global_toast_list_single_toast_dismissed',
+  ALL_TOASTS_DISMISSED = 'global_toast_list_all_toasts_dismissed',
+}
+
+export enum FieldType {
+  RECURRENCE_COUNT = 'toast_deduplication_count',
+  TOAST_MESSAGE = 'toast_message',
+}
+
+const fields: Record<FieldType, RootSchema<Record<string, unknown>>> = {
+  [FieldType.TOAST_MESSAGE]: {
+    [FieldType.TOAST_MESSAGE]: {
+      type: 'text',
```

Comment:
Using the `text` type means we can search for records against a matching query, but we wouldn't be able to aggregate - such as getting a breakdown of the top messages or finding the most occurrent message.

Maybe this should use `keyword`?